libres/lib/config/config_schema_item.cpp

```diff
@@ -66,398 +66,465 @@ typedef struct validate_struct validate_type;
-          config_error_add( error_list , util_alloc_sprintf("%s: is not a valid value for: %s.",stringlist_iget( token_list , iarg + 1) , item->kw));
-          OK = false;
-        }
-      }
-    } else if (item->validate->indexed_selection_set.size()) {
-      for (int iarg = 0; iarg < argc; iarg++) {
-        if ((item->validate->argc_max > 0) || (iarg < item->validate->argc_min)) {  /* Without this test we might go out of range on the indexed selection set. */
-          const auto * selection_set = validate_iget_selection_set( item->validate , iarg);
-          if (selection_set && selection_set->size()) {
-            if (!selection_set->count( stringlist_iget(token_list, iarg + 1))) {
-              config_error_add( error_list , util_alloc_sprintf("%s: is not a valid value for item %d of \'%s\'.",stringlist_iget( token_list , iarg + 1) , iarg + 1 , item->kw));
-              OK = false;
+    if (OK) {
+        /* Validating selection set - first common, then indexed */
+        if (item->validate->common_selection_set.size()) {
+            for (int iarg = 0; iarg < argc; iarg++) {
+                if (!item->validate->common_selection_set.count(
+                        stringlist_iget(token_list, iarg + 1))) {
+                    config_error_add(error_list,
+                                     util_alloc_sprintf(
+                                         "%s: is not a valid value for: %s.",
+                                         stringlist_iget(token_list, iarg + 1),
+                                         item->kw));
+                    OK = false;
+                }
+            }
+        } else if (item->validate->indexed_selection_set.size()) {
+            for (int iarg = 0; iarg < argc; iarg++) {
+                if ((item->validate->argc_max > 0) ||
+                    (iarg <
+                     item->validate
+                         ->argc_min)) { /* Without this test we might go out of range on the indexed selection set. */
+                    const auto *selection_set =
+                        validate_iget_selection_set(item->validate, iarg);
+                    if (selection_set && selection_set->size()) {
+                        if (!selection_set->count(
+                                stringlist_iget(token_list, iarg + 1))) {
+                            config_error_add(
+                                error_list,
+                                util_alloc_sprintf(
+                                    "%s: is not a valid value for item %d of "
+                                    "\'%s\'.",
+                                    stringlist_iget(token_list, iarg + 1),
+                                    iarg + 1, item->kw));
+                            OK = false;
+                        }
+                    }
+                }
             }
-          }
         }
-      }
-    }
 
-    /*
+        /*
```

Comment:
Indentation of comment now looks weird because the indentation of the code has changed.